FlexMeasures/flexmeasures
# feat: add redis liveness probe

## Description
Check the health of redis instance and expose it trough the API endpoint `/health/ready`

## Look & Feel

### Request
```
[GET] http://localhost:5000/api/v3_0/health/ready
```

### Response
```json
{
	"database_redis": false,
	"database_sql": true,
	"status": 503
}
```

## Testing Steps


1) Configure credentials

```terminal
echo 'FLEXMEASURES_REDIS_PASSWORD="MyPassWord"' >> ~/.flexmeasures.cfg 
```

2) Run FM Server

```terminal
flexmeasures run
```

3) Get and check status

```terminal
curl localhost:5000/api/v3_0/health/ready
```

> :gear: Check the response (status=503 and database_redis=false)
>```json
>{
>	"database_redis": false,
>	"database_sql": true,
>	"status": 503
>}
>```

4) Create Redis instance with Docker

```terminal
sudo docker run --name my-redis -p 6379:6379 -d redis --requirepass "MyPassWord"
```

5) Get and check status

```terminal
curl localhost:5000/api/v3_0/health/ready
```

> :gear: Check the response (status=200 and database_redis=true)
>```json
>{
>	"database_redis": true,
>	"database_sql": true,
>	"status": 200
>}
>```

Clean up

```terminal
sudo docker stop my-redis && sudo docker rm my-redis
```

---

Closes #663
